### DISCUSSION TITLE
Downloading a large file causes the memory to be fully occupied

### DISCUSSION THREAD

**Discussion by liuhao111418** [mitmproxy/mitmproxy#7079] [Q&A]
#### Problem Description
When downloading large files, they are constantly stored in memory, causing the memory to grow until it is full

#### Steps to reproduce the behavior:
1. Enable mitmproxy
2. Download large files through browser
3. The memory keeps growing, and excessive files can cause the memory to fill up

#### System Information
Mitmproxy: 10.4.0
Python:    3.12.3
OpenSSL:   OpenSSL 3.2.2 4 Jun 2024
Platform:  Windows-10-10.0.19044-SP0

**Comment by liuhao111418** [comment]
If I encounter such a large file download, how should I determine and release the traffic of this file, and directly return it to the browser

**Comment by mhils** [comment]
https://docs.mitmproxy.org/stable/overview-features/#streaming
